signal_handler.c

```diff
@@ -35,6 +35,15 @@
 # include <stdint.h>
 #endif
 
+#if PHP_MAJOR_VERSION < 7
+#define _DECLARE_ZVAL(name) zval * name
+#define hp_ptr_dtor(val) zval_ptr_dtor(&val);
```

Comment:
If you add a couple defines to the <7 block here, you can simplify your usages below.

```
#define zend_release_string(str) efree(str)
#define ZSTR_VAL(str) (str)
```

This is sort of a cheat, which relies on your using the same name in PHP5 for a char* as you'd use in PHP7 for a zend_string*, but it'll make your later migration to PHP7-only simpler.

Or, you could define new names (e.g. hp_release_string() and hp_strval()) and point those at the approproiate PHP5/PHP7 versions.

This whole comment is non-essential though, and is only aimed at making the majority of the source more readable by not littering #ifdef statements all over the place.